src/main/java/edu/cornell/kfs/module/purap/rest/resource/CuEinvoiceApiResource.java

```diff
@@ -0,0 +1,245 @@
+package edu.cornell.kfs.module.purap.rest.resource;
+
+import com.google.gson.Gson;
+import edu.cornell.kfs.module.purap.CUPurapConstants;
+import edu.cornell.kfs.module.purap.dataaccess.impl.CuCuEinvoiceDaoOjb;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.kuali.kfs.krad.dao.LookupDao;
+import org.kuali.kfs.krad.util.KRADPropertyConstants;
+import org.kuali.kfs.module.purap.PurapPropertyConstants;
+import org.kuali.kfs.module.purap.businessobject.PurApItem;
+import org.kuali.kfs.module.purap.document.PurchaseOrderDocument;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.kfs.vnd.VendorConstants;
+import org.kuali.kfs.vnd.businessobject.VendorAddress;
+import org.kuali.kfs.vnd.businessobject.VendorDetail;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+@Path("api")
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+public class CuEinvoiceApiResource {
+
+    private static final Log LOG = LogFactory.getLog(CuEinvoiceApiResource.class);
+
+    private LookupDao lookupDao;
+    private Gson gson = new Gson();
```

Comment:
I'd be fine with switching to Jackson as long as it would be as simple as Gson, meaning just one line of code (toJson)

Reply:
I believe Jackson also supports converting Properties into JSON, but I am not certain if it can perform the one-line conversion like GSON can. Given that KFS has both the Jackson and GSON libraries at this time, I'm fine with this code using whichever library makes the most sense to you.